src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -569,22 +645,18 @@ CreateOffer::applyGuts (ApplyView& view, ApplyView& view_cancel)
     bool const bFillOrKill (uTxFlags & tfFillOrKill);
     bool const bSell (uTxFlags & tfSell);
 
-    STAmount saTakerPays = tx().getFieldAmount (sfTakerPays);
-    STAmount saTakerGets = tx().getFieldAmount (sfTakerGets);
+    auto saTakerPays = tx()[sfTakerPays];
```

Comment:
Ok, this is why I'm confused: https://github.com/ximinez/rippled/commit/c15394c42ad19cb6dee761cf63bb04738b7b5316#diff-afcc1209f06865ece476ddeaee53c363R82